hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SerializationServiceV1.java

```diff
@@ -181,6 +184,10 @@ private void registerJavaTypeSerializers() {
         registerConstant(Enum.class, new EnumSerializer());
         registerConstant(ArrayList.class, new ArrayListStreamSerializer());
         registerConstant(LinkedList.class, new LinkedListStreamSerializer());
+        MapStreamSerializer mapSerializer = new MapStreamSerializer();
+        registerConstant(HashMap.class, mapSerializer);
+        registerConstant(ConcurrentHashMap.class, mapSerializer);
```

Comment:
If we want to guarantee the type will be same also, using the same serializer will not work. 
ArrayList and LinkedList serializer are different because of that. 
That being said, we may reconsider the concurrent List type support also. Having to register all concrete types does not seem maintainable to me. 
Why can't we just register Map.class, List.class ?

Reply:
My 2 cents - we can support only the main couple of implementations. The user can always add custom serialisation for any non-typical collection implementation but I still think that if I provide a `ConcurrentHashMap` while serialising, I would expect the same when deserialising (otherwise subtle concurrency bugs may arise).